composer/loggers/logger_destination.py

```diff
@@ -38,29 +38,40 @@ class LoggerDestination(Callback, ABC):
             Batch 0: {'rank_zero_seed': ...}
     """
 
-    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
-        """Log data.
+    def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
+        """Log hyperparameters, configurations, and settings.
 
-        Subclasses should implement this method to store logged data (e.g. write it to a file, send it to a server,
-        etc...). However, not all loggers need to implement this method.
+        Logs any parameter/configuration/setting that doesn't vary during the run.
 
-        .. note::
+         Args:
+            hyperparameters (Dict[str, Any]): A dictionary mapping hyperparameter names
+                (strings) to their values (Any).
+        """
+        del hyperparameters  # unused
+        pass
 
-            This method will block the training loop. For optimal performance, it is recommended to deepcopy the
-            ``data`` (e.g. ``copy.deepcopy(data)``), and store the copied data in queue. Then, either:
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+        """Log metrics or parameters that vary during training.
 
-            *   Use background thread(s) or process(s) to read from this queue to perform any I/O.
-            *   Batch the data together and flush periodically on events, such as
-                :attr:`.Event.BATCH_END` or :attr:`.Event.EPOCH_END`.
+        Args:
+            metrics (Dict[str, float]): Dictionary mapping metric name (str) to metric
+                scalar value (float)
+            step (Optional[int], optional): The current step or batch of training at the
+                time of logging. Defaults to None. If not specified the specific
+                LoggerDestination implementation will choose a step (usually a running
+                counter).
+        """
+        del metrics, step  # unused
+        pass
 
-                .. seealso:: :class:`~composer.loggers.file_logger.FileLogger` as an example.
+    def log_traces(self, traces: Dict[str, Any]):
```

Comment:
Not sure if this is actually necessary?

Reply:
`log_traces` was added, so that wandb, tb, cometml don't have to log the engine traces which are verbose and only useful if you're using the filelogger or the console logger